Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/Identity.cs

```diff
@@ -23,42 +34,61 @@ public void AssignTierQuotaFromDefinition(TierQuotaDefinition definition)
         _tierQuotas.Add(tierQuota);
     }
 
-    public async Task UpdateMetrics(IEnumerable<string> metrics, IMetricCalculatorFactory factory, CancellationToken cancellationToken)
+    public async Task UpdateMetrics(IEnumerable<MetricKey> metrics, MetricCalculatorFactory factory, CancellationToken cancellationToken)
     {
         foreach (var metric in metrics)
         {
             var metricCalculator = factory.CreateFor(metric);
-            await UpdateMetricAsync(metric, metricCalculator, cancellationToken);
+            await UpdateMetric(metric, metricCalculator, cancellationToken);
         }
-        return;
     }
 
-    private async Task UpdateMetricAsync(string metric, IMetricCalculator metricCalculator, CancellationToken cancellationToken)
+    private void UpdateMetricStatus(MetricKey metricKey, IEnumerable<Quota> quotasFromMetric)
     {
-        if (TierQuotas.Count == 0)
+        var metricStatus = _metricStatuses.FirstOrDefault(m => m.MetricKey == metricKey);
```

Comment:
Makes sense. We're not addressing the MetricStatus entity in regard to its database representation and that has become an issue because the application will not run properly. I have invested quite some time in creating a migration but I keep facing EF errors. Anyhow, it seems to me that it is unclear how we're to store this on the database. To me, it would make sense that there would be a composite key with the Identity (Address) and the MetricKey. Is that how we'll proceed?

Sorry for highjacking the post, but this matter has been bugging me.

Reply:
> We're not addressing the MetricStatus entity in regard to its database representation

We're not? But that should be part of this PR.

> To me, it would make sense that there would be a composite key with the Identity (Address) and the MetricKey.
Yes, that's what it should look like.